src/core/lib/lattice/hal/default/dcrtpoly.cpp

```diff
@@ -1376,6 +1379,29 @@ std::shared_ptr<typename DCRTPolyImpl<VecType>::Params> DCRTPolyImpl<VecType>::G
     return std::make_shared<DCRTPolyImpl::Params>(2 * this->GetRingDimension(), moduliQP, rootsQP);
 }
 
+template <typename VecType>
+void DCRTPolyImpl<VecType>::TimesQovert(
+    const std::shared_ptr<DCRTPolyImpl::Params> paramsQ,
+    const std::vector<NativeInteger> &tInvModq,
+    const std::vector<NativeInteger> &tInvModqPrecon,
+    const NativeInteger &t,
+    const NativeInteger &NegQModt,
+    const NativeInteger &NegQModtPrecon) {
+  usint sizeQ = m_vectors.size();
+  usint ringDim = this->GetRingDimension();
+#pragma omp parallel for
+  for (size_t i = 0; i < sizeQ; i++) {
```

Comment:
we should make sure that the size of m_vectors, tInvModqPrecon, tInvModq and paramsQ->GetParams() is the same